plugins/inputs/whois/whois.go

```diff
@@ -0,0 +1,223 @@
+			w.Log.Errorf("WHOIS query failed for %s: %v", domain, err)
+			acc.AddError(err)
+
+			// Always register a metric, even on failure
+			acc.AddFields("whois", map[string]interface{}{
+				"status": 0, // Mark failure
+			}, map[string]string{
+				"domain":        domain,
+				"domain_status": "UNKNOWN",
+			})
+			continue
+		}
+
+		// Parse WHOIS data using whois-parser
+		parsedWhois, err := w.ParseWhoisData(rawWhois)
+		if err != nil {
+			w.Log.Errorf("WHOIS parsing failed for %s: %v", domain, err)
+			acc.AddError(err)
+
+			// Always register a metric, even on failure
+			acc.AddFields("whois", map[string]interface{}{
+				"status": 0, // Mark failure
+			}, map[string]string{
+				"domain":        domain,
+				"domain_status": "UNKNOWN",
+			})
+			continue
+		}
+
+		w.Log.Debugf("Parsed WHOIS data for %s: %+v", domain, parsedWhois)
+
+		// Prevent nil pointer panic
+		if parsedWhois.Domain == nil {
+			w.Log.Warnf("No domain info found for %s", domain)
+
+			// Always register a metric, even on failure
+			acc.AddFields("whois", map[string]interface{}{
+				"status": 0, // Mark failure
+			}, map[string]string{
+				"domain":        domain,
+				"domain_status": "UNKNOWN",
+			})
+			continue
+		}
+
+		// Extract expiration date
+		expiration := parsedWhois.Domain.ExpirationDateInTime
+		if expiration == nil {
+			w.Log.Warnf("No expiration date found for %s", domain)
+			continue
+		}
+
+		// Extract creation date
+		created := parsedWhois.Domain.CreatedDateInTime
+		if created == nil {
+			w.Log.Warnf("No created date found for %s", domain)
+			continue
+		}
+
+		// Extract updated date
+		updated := parsedWhois.Domain.UpdatedDateInTime
+		if updated == nil {
+			w.Log.Warnf("No updated date found for %s", domain)
+			continue
+		}
+
+		// Extract DNSSEC status
+		dnssec := parsedWhois.Domain.DNSSec
+
+		// Extract NameServers status
+		nameServers := parsedWhois.Domain.NameServers
+		if len(nameServers) == 0 {
+			w.Log.Warnf("No name servers found for %s", domain)
+			continue
+		}
+
+		// Extract registrar name (handle nil)
+		registrar := ""
+		if parsedWhois.Registrar != nil {
+			registrar = parsedWhois.Registrar.Name
+		}
+
+		// Extract status (handle nil)
+		domainStatus := "UNKNOWN"
+		if parsedWhois.Domain.Status != nil {
+			domainStatus = simplifyStatus(parsedWhois.Domain.Status)
+		}
+
+		// Calculate expiry in seconds
+		expiry := int(expiration.Sub(now).Seconds())
+
+		// Add metrics
+		fields := map[string]interface{}{
+			"creation_timestamp":   created.Unix(),
+			"dnssec_enabled":       boolToInt(dnssec),
+			"expiration_timestamp": expiration.Unix(),
+			"expiry":               expiry,
+			"updated_timestamp":    updated.Unix(),
+			"registrar":            registrar,
+			"domain_status":        domainStatus,
+			"status":               1,
+		}
+		tags := map[string]string{
+			"domain": domain,
+		}
+
+		if w.IncludeNameServers {
+			fields["name_servers"] = strings.Join(nameServers, ",")
+		}
```

Comment:
Users can use `fieldexclude` to not have this field, it doesn't need to be a configuration option.

It could have been a config option, if that would resulted in less data to be queried from the WHOIS server.